src/profile-logic/process-profile.js

```diff
@@ -1379,6 +1379,64 @@ function processMarkerSchema(geckoProfile: GeckoProfile): MarkerSchema[] {
   return combinedSchemas;
 }
 
+export function insertExternalMarkersIntoProfile(
+  markers: MixedObject,
+  geckoProfile: GeckoProfile
+): void {
+  console.log('DEBUG FLO', markers);
+  console.log('DEBUG FLO', 'profile', geckoProfile);
+  for (const schema of markers.markerSchema) {
+    geckoProfile.meta.markerSchema.push(schema);
+  }
+
+  const categoryMap = new Map();
+  for (let i = 0; i < markers.categories.length; ++i) {
+    const cat = markers.categories[i];
+    let index = geckoProfile.meta.categories.findIndex(
+      (c) => c.name === cat.name
+    );
+    if (index === -1) {
+      index = geckoProfile.meta.categories.push(cat) - 1;
+    }
+    categoryMap.set(i, index);
+  }
+
+  const firstThread = geckoProfile.threads[0];
```

Comment:
Hmm I don't think this first thread is guaranteed to have the parent process main thread all the time. Since the profile collection is async, whichever process finishes first ends up the first thread, which is 99% of the time the parent process. But it's theoretically possible to have another.

Probably you want to do something like this:
https://github.com/firefox-devtools/profiler/blob/692bcb8f05f00433f514f028c1f2b4058f7297da/src/profile-logic/process-profile.js#L1817-L1820